pkg/test/framework/scope.go

```diff
@@ -91,12 +93,6 @@ func (s *scope) reset() error {
 		}
 	}
 
-	if s.parent != nil {
```

Comment:
Not sure why you're removing this?

Reply:
This won't work for subtests running in parallel. And I'm not sure it's correct regardless.  This effectively bubbles all the way up the resource chain and resets the entire system.